server/leafnode.go

```diff
@@ -1764,6 +1765,13 @@ func (c *client) processLeafNodeConnect(s *Server, arg []byte, lang string) erro
 		return err
 	}
 
+	// Reject a cluster that contains spaces or line breaks.
+	if proto.Cluster != _EMPTY_ && strings.ContainsFunc(proto.Cluster, unicode.IsSpace) {
```

Comment:
Thanks Ivan, we eventually went back into that direction to avoid breaking changes on the patch release.
Adding the check on INFO would make leafnodes prevent being able to connect to clusters/gateways names that have spaces which although we would be disallowing in v2.11, does not seem to cause issues so leaving that for v2.10 for now.
About the server name being reused for the cluster name, added that as a test here but I think we could add a warning during the config check?
https://github.com/nats-io/nats-server/blob/3833649378d36a407b045adeb288c534465a0208/server/jetstream_cluster_4_test.go#L2104-L2109

Reply:
I was just saying that users may get failure to connect but may not understand that it is due to the server name, since they did not specify a cluster name. Maybe [here](https://github.com/nats-io/nats-server/blob/92a857907237f1d35ab1ebc7c42e3bd41cd76008/server/server.go#L744) when we use the server name, you could add the check and indeed issue a warning if it contains spaces/new_lines?

Reply:
Updated for it to show a warning:

```
[61124] 2024/08/06 06:04:13.218801 [INF] Starting nats-server
[61124] 2024/08/06 06:04:13.219424 [INF]   Version:  2.10.19-RC.2
[61124] 2024/08/06 06:04:13.219428 [INF]   Git:      [not set]
[61124] 2024/08/06 06:04:13.219431 [INF]   Cluster:  L E A F 2
[61124] 2024/08/06 06:04:13.219433 [INF]   Name:     L E A F 2
[61124] 2024/08/06 06:04:13.219435 [INF]   ID:       NCYPTW4H3TEKDHA2H46FCM6WTX7GOSSTQ3LALGKBUTUPGZ4P7XH37GSD
[61124] 2024/08/06 06:04:13.219466 [INF] Using configuration file: leaf.conf
[61124] 2024/08/06 06:04:13.220748 [INF] Starting http monitor on 127.0.0.1:56401
[61124] 2024/08/06 06:04:13.220925 [INF] Listening for leafnode connections on 127.0.0.1:56402
[61124] 2024/08/06 06:04:13.221059 [INF] Advertise address for leafnode is set to 127.0.0.1:56402
[61124] 2024/08/06 06:04:13.221403 [WRN] Server name has spaces and used as the cluster name, leaf remotes may not connect properly
[61124] 2024/08/06 06:04:13.221486 [INF] Listening for client connections on 127.0.0.1:56403
```